os/drivers/kernel_test/kernel_test_drv.c

```diff
@@ -153,13 +153,17 @@ static int kernel_test_drv_ioctl(FAR struct file *filep, int cmd, unsigned long
 			* which belongs to any other app in the system. Here,
 			* we choose to return the address of the app heap
 			*/
-			uint32_t binid = sched_self()->group->tg_binidx;
-			binid = (binid + 1) % (binary_manager_get_ucount() + 1);
-			if (binid == 0) {
-				binid++;
+			uint32_t binidx = sched_self()->group->tg_binidx;
+			binidx = (binidx + 1) % (binary_manager_get_ucount() + 1);
+			if (binidx == 0) {
+				binidx++;
+			}
+			struct tcb_s *tcb = (struct tcb_s *)sched_gettcb(BIN_ID(binidx));
```

Comment:
It looks that we want to set `obj->addr` as its user heap address.
Every tcbs in a binary has its user heap address.
So I think we don't need to find the parent.
We can get the heap address with its own like below.
```
struct tcb_s *tcb = sched_self();
if (tcb) {
    obj->addr = (volatile uint32_t *)tcb->uheap;
}
```

Reply:
It is used to get the heap address of another binary so it gets a index of another binary here.
You can find the usage in mpu_test.c:58 and recovery:67.